internal/broker/broker.go

```diff
@@ -540,6 +543,16 @@ func (b *Broker) CancelIsAuthenticated(sessionID string) {
 	}
 }
 
+// UserPreCheck checks if the user is valid and can be allowed to authenticate.
+func (b *Broker) UserPreCheck(username string) error {
+	for _, suffix := range b.allowedSSHSuffixes {
+		if strings.HasSuffix(username, suffix) {
+			return nil
+		}
+	}
+	return errors.New("username does not match the allowed suffixes")
```

Comment:
How is then this error transcribed to ssh?

Reply:
It's not. This error goes to NSS, but we don't show it anywhere else besides the system logs.

Reply:
ok, sounds good to me then!